Notification component

karelhala opened the pull request:
New notification component, change of data-tables in demo application, dataTable service now stores data which are recieved from server, this way it's easier to sort and limit data.

New tile component which is for displaying small tile with icon and name and big one for data and icon.

karelhala committed:
Add Rx, new component for alerts, add them to demo
Rx is taking care for assynchronous events, especially alerts, which are registered in Subject and called on Subject.
Alerts itself are then inserted with new component, where you can specify how many of them will be visible and if you want, after how much time they will dissapear.
Inserting this component to demo as well, so we can see how they behave

karelhala committed:
Fix notification services to work more with Rxjs
change RXjs version to 4, since we use currently Angular 1.5.x and this version uses only Rx 4. After moving to Angular 2, we can change to Rx 5 easily.

karelhala committed:
Add tile view component, sort items
new component for tile views, this component has 2 usage, small without data and big with data.
Another component for sorting items

karelhala committed:
Update items after they are sorted
In datables there was problem that whenever sort or limit was used data were not refreshed properly.

karelhala committed:
Fix typescript errors and warnings

karelhala committed:
Update styles for small icons

mtho11 commented on webpack.config.js:
```diff
@@ -32,6 +32,9 @@ var webpack = require('webpack'),
         selector: '[ng-app]'
       })
     }),
+    new webpack.ProvidePlugin({
```
I'm not really wild about the webpack ProvidePlugin, since it magically provides that module everywhere. Kind of similiar to global variables but for modules. Explicitly importing Rx seems like a better way. Most components won't need to have the whole Rx module any but a very specific subset of operators.
Tooling that recognizes imports probably won't understand implicit webpack modules and will show errors all over the place (possibly even for linting tools).

mtho11 commented:
LGTM. wdyt @ammendonca @abonas 

Jenkins build passed but travis build failed. I'll leave the travis issues to you @karelhala

mtho11 commented:
@karelhala any luck with Travis?

abonas commented:
it's really hard to review such a big PR. some commits imo are not relevant to the new component?

abonas commented:
it will also help to have screenshots in the PR to understand where/what is new

karelhala commented:
@mtho11  even Jenkins had problem, wrong build script. I forgot to add npm install to it... with travis is problem with npm install which is odd.

@abonas I'll work on demo pages for these components and attach screenshots to PR, good point.

abonas commented:
I've pulled the branch and 'npm install', I have tones of failures, but I guess it's due to some wrong versions of components that I have locally?
(I have node v5.10.1)

bufferutil@1.2.1 install /home/abonas/hawkular/hawkular-ui-components/node_modules/bufferutil

> node-gyp rebuild

make: Entering directory `/home/abonas/hawkular/hawkular-ui-components/node_modules/bufferutil/build'
  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
In file included from ../src/bufferutil.cc:16:
../../nan/nan.h:43:3: error: #error This version of node/NAN/v8 requires a C++11 compiler
In file included from ../src/bufferutil.cc:7:

karelhala commented:
I have same version of node so this should not be the problem. I have removed all node modules and tried reinstall and still no error. Will go deeply trough this, demo page is done so one I'll figure out what causes npm install to throw these errors I'll push.

karelhala commented:
Looks like problem is with gcc/g++ compiler, strange thing is that I did not installed any newer compiler and it works for me. However based on this post [Building a Node.js project](https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Node.js-v4-%28or-io.js-v3%29-compiler-requirements) we have to specify gcc/g++ compiler to something newer. Another solution might be to use different build environment (it's called trusty). I'll go with the newer environment and see what will be changed.

However if you @abonas have problem with this I think installing new gcc/g++ compiler is needed, you can run `g++ --version` to see what version are you using and if it's lower than 4.8 that is the problem.

karelhala committed:
Add tile view to demo page
new view and component was created, so let's add them to demo application to showcase these changes. Also update table to correspond with new changes.

Let's use different build environment, since the last one had problems with g++ compiler, Trusty environment should be using newer version.

karelhala commented:
Ok, now the travis works, but not jenkins...

abonas commented:
@karelhala thanks for the thorough investigation.
the version I have is 4.4.7, so I'll see if upgrade is possible

karelhala commented:
test this please

karelhala commented:
retest this please

karelhala commented:
test this please

karelhala committed:
Add secret and remove --force
when merging PR secret has to be used in hashed form. Also remove --force from publishing scripts

karelhala commented:
I am unable to force jenkins into pushing changes to gh-pages. Will look to it tomorrow.

abonas commented:
@karelhala it looks like I can't upgrade gcc, but it's not related to this PR, of course

mtho11 commented:
Merged #129
